apps/aefate/src/aefa_chain_api.erl

```diff
@@ -194,25 +198,65 @@ traverse_to_key_hash(H, KeyHash) ->
         _Height -> traverse_to_key_hash(H, aec_headers:prev_key_hash(Header))
     end.
 
--spec contract_fate_code(pubkey(), state()) -> 'error' |
-                                               {'ok', {term(), aect_contracts:vm_version()}, state()}.
-contract_fate_code(Pubkey, #state{primop_state = PState} = S) ->
+-spec put_contract(aect_contracts:contract(), state()) -> state().
+put_contract(Contract, #state{primop_state = PS0} = S) ->
```

Comment:
You know it's possible to predict a contract address without deploying it and send money to that account?
1. Foo pre-calculates the address of Bar - X 
2. Foo sends funds to address X
3. Foo deploys Bar at address X(it's enough to predict the nonces)

Your code would nuke the account balance.
BTW. @hanssv @UlfNorell How does the node handle this case generally in create_tx? Should the tokens be lost or still present after deploy?

Reply:
The code does not consider `payable`/`non_payable`

But I wonder if this code should not use `aeprimop` accessing `aeprimop_state` directly is not right?

Reply:
Should work now. I fix the accounts' payability on collision and I leave the account on init fail – just in case it was there before.